hq/test/logic/IamRemediationTest.scala

```diff
@@ -144,7 +145,39 @@ class IamRemediationTest extends FreeSpec with Matchers {
   }
 
   "lookupCredentialId" - {
-    "TODO" ignore {}
+    val nonMatchingAccessKey = CredentialMetadata("adam.fisher", "AKIAIOSFODNN1EXAMPLE", date.minusDays(1), CredentialActive)
+    val matchingAccessKey = CredentialMetadata("amina.adewusi", "AKIAIOSFODNN2EXAMPLE", date, CredentialActive)
+    val matchingAccessKey2 = CredentialMetadata("amina.adewusi", "AKIAIOSFODNN3EXAMPLE", date, CredentialActive)
+
+    "given a key creation date matches a date in the metadata, return the correct metadata" in {
+      val result = lookupCredentialId(date, List(matchingAccessKey, nonMatchingAccessKey))
+      result.value.username shouldEqual matchingAccessKey.username
+    }
+    "given there are no matching key creation dates, return a failure" in {
+      val result = lookupCredentialId(date, List(nonMatchingAccessKey))
+      result.isFailedAttempt() shouldBe true
+    }
+    // both user's access keys sharing the exact same date is an edge case, because the creation date is accurate to the second
+    // and it's unlikely both keys would be created at exactly the same time, but could happen, especially if created using the CLI.
+    // If this happens then we won't know how to identify the keys' id, which is required to disable it, so we return a Failure.
+    "given a key creation date matches two dates in the metadata, return a failure" in {
+      val result = lookupCredentialId(date, List(matchingAccessKey2, matchingAccessKey))
+      result.isFailedAttempt() shouldBe true 
+    }
+    // AWS's credential's report shows that a key's last rotated date is recorded to the second,
+    // so this function must seek to match to the dates up to the second, but not the millisecond.
+    "given a key creation date matches to the minute, but not the second, return a failure" in {
+      val keyCreationDate = new DateTime(1,1,1,1,1,1)
```

Comment:
Careful about using DateTimes with really funny years, like this! It'll be fine here (I think) but depending how the date gets used / represented as it flows through the program, this could break things! If the Date is ever converted to a (32-bit) timestamp, it won't be possible to represent the year `1` and this will either get truncated to the first of January 1970, or something even more surprising than that will happen!

Best to use a "realistic" date, even for test data (just like the one at the top of the file). Any time in the decades around now is fine.

Reply:
Sorry, this was a mistake! Thanks!